lib/mqtt/listeners/attach_listener.dart

```diff
@@ -16,18 +24,58 @@ class AttachListener {
       throw Exception("Device ID is null, cannot set listeners!");
     }
     mqttClient.addListeners({
-      "$baseTopic/devices/$deviceId/attach": handleAttach,
-      "$baseTopic/devices/$deviceId/detach": handleDetach,
+      "$baseTopic/devices/$deviceId/attached": handleAttach,
+      "$baseTopic/devices/$deviceId/detached": handleDetach,
     });
   }
 
   /// Callback function for handling attach messages
-  void handleAttach(String message) {
-    SimpleLogger().info("Handling attach message: $message");
+  void handleAttach(String message) async {
+    SimpleLogger().info("Handling attach message...");
+    final attachedMessage =
+        AbstractListener.decodeMessage<MqttDeviceAttachedToExhibition>(
+            message, MqttDeviceAttachedToExhibition);
+    SimpleLogger().info(attachedMessage);
+
+    // There's no support for having device attached to multiple exhibitions and therefore existing exhibitions are deleted
+    await pageDao.deletePages();
+    await layoutDao.deleteLayouts();
+    await exhibitionDao.deleteExhibitions();
```

Comment:
exhibition entity should be named more appropriately, it does not contain details from exhibition, it contains details how the device is connected to an exhibition, so more approperiate name would be e.g. DeviceExhibitionDetails

Reply:
Resolved.